node-admin/src/main/java/com/yahoo/vespa/hosted/node/admin/util/ConfigServerHttpRequestExecutor.java

```diff
@@ -13,90 +13,59 @@
-        }
+    private SelfCloseableHttpClient client;
 
-        super.finalize();
-    }
-
-    public static ConfigServerHttpRequestExecutor create(Collection<URI> configServerUris) {
-        PoolingHttpClientConnectionManager cm = new PoolingHttpClientConnectionManager(getConnectionSocketFactoryRegistry());
-        cm.setMaxTotal(200); // Increase max total connections to 200, which should be enough
-
-        // Have experienced hang in socket read, which may have been because of
-        // system defaults, therefore set explicit timeouts. Set arbitrarily to
-        // 15s > 10s used by Orchestrator lock timeout.
-        int timeoutMs = 15_000;
-        RequestConfig requestBuilder = RequestConfig.custom()
-                .setConnectTimeout(timeoutMs) // establishment of connection
-                .setConnectionRequestTimeout(timeoutMs) // connection from connection manager
-                .setSocketTimeout(timeoutMs) // waiting for data
-                .build();
-
-        return new ConfigServerHttpRequestExecutor(randomizeConfigServerUris(configServerUris),
-                                                   HttpClientBuilder.create()
-                                                           .setDefaultRequestConfig(requestBuilder)
-                                                           .disableAutomaticRetries()
-                                                           .setUserAgent("node-admin")
-                                                           .setConnectionManager(cm).build());
-    }
+    public static ConfigServerHttpRequestExecutor create(
+            Collection<URI> configServerUris, Optional<KeyStoreOptions> keyStoreOptions, Optional<KeyStoreOptions> trustStoreOptions) {
+        Supplier<SelfCloseableHttpClient> clientSupplier = () -> createHttpClient(keyStoreOptions, trustStoreOptions);
+        ConfigServerHttpRequestExecutor requestExecutor = new ConfigServerHttpRequestExecutor(
+                randomizeConfigServerUris(configServerUris), clientSupplier.get());
 
-    private static Registry<ConnectionSocketFactory> getConnectionSocketFactoryRegistry() {
-        try {
-            SSLConnectionSocketFactory sslSocketFactory = new SSLConnectionSocketFactory(
-                    new SSLContextBuilder().loadTrustMaterial(null, (arg0, arg1) -> true).build(),
-                    NoopHostnameVerifier.INSTANCE);
-
-            return RegistryBuilder.<ConnectionSocketFactory>create()
-                    .register("http", PlainConnectionSocketFactory.getSocketFactory())
-                    .register("https", sslSocketFactory)
-                    .build();
-        } catch (GeneralSecurityException e) {
-            throw new RuntimeException("Failed to create SSL context", e);
+        if (keyStoreOptions.isPresent() || trustStoreOptions.isPresent()) {
+            requestExecutor.clientRefresherScheduler.scheduleAtFixedRate(() -> requestExecutor.client = clientSupplier.get(),
```

Comment:
client must be volatile and it should be documented why it has to

Reply:
Fixed.